sample/android/src/main/java/co/nimblehq/kmm/template/ui/screens/home/HomeViewModel.kt

```diff
@@ -0,0 +1,30 @@
+package co.nimblehq.kmm.template.ui.screens.home
+
+import androidx.lifecycle.viewModelScope
+import co.nimblehq.kmm.template.domain.usecases.UseCase
+import co.nimblehq.kmm.template.ui.base.BaseViewModel
+import co.nimblehq.kmm.template.ui.models.UiModel
+import co.nimblehq.kmm.template.ui.models.toUiModel
+import co.nimblehq.kmm.template.util.DispatchersProvider
+import kotlinx.coroutines.flow.*
+
+class HomeViewModel(
+    dispatchersProvider: DispatchersProvider,
+    useCase: UseCase,
+) : BaseViewModel() {
+
+    private val _uiModels = MutableStateFlow<List<UiModel>>(emptyList())
+    val uiModels: StateFlow<List<UiModel>> = _uiModels
+
+    init {
+        useCase()
+            .injectLoading()
+            .onEach { result ->
+                val uiModels = result.map { it.toUiModel() }
+                _uiModels.emit(uiModels)
+            }
+            .flowOn(dispatchersProvider.io)
+            .catch { e -> _error.emit(e) }
```

Comment:
Optional:
```suggestion
            .catch { exception -> _error.emit(exception) }
```

Reply:
@lydiasama We can not change this from KMM as we're reusing it from the Android Templates. Let's improve it there first 😄

Reply:
😢